sympy/core/evalf.py

```diff
@@ -82,7 +93,13 @@ class PrecisionExhausted(ArithmeticError):
 """
 
 
-def fastlog(x):
+MPF_VALUE_TUPLE = tTuple[int, int, int, int]
+SCALED_ZERO_TUPLE = tTuple[List[int], int, int, int]
+TEMP_RESULT = Any
```

Comment:
These names are quite long and I'm not sure what they really mean. If we're going to come up with names for things like this then they need to be informative names and there should be some comments here explaining what they mean. The name `TEMP_RESULT` does not convey much information to me and the fact that it is an alias for `Any` means that it gives the appearance of a meaningful type hint when in fact the hint says nothing.

Reply:
For `MPF_VALUE_TUPLE` and `TEMP_RESULT`, I didn't come up with them randomly, they are just below the comment https://github.com/sympy/sympy/blob/9019765edb0d0b4bbd29674a58b93ef4e75b92f8/sympy/core/evalf.py#L60-L83 which already "define" these names.

Reply:
For `SCALED_ZERO_TUPLE`, it's used in the functions related to `scaled_zero` and can be found at https://github.com/sympy/sympy/blob/9019765edb0d0b4bbd29674a58b93ef4e75b92f8/sympy/core/evalf.py#L152-L156

Reply:
`TEMP_RESULT` is `Any` because I failed to find a way that makes mypy happy. I originally use the type `tUnion[tTuple[Optional[MPF_VALUE_TUPLE], Optional[MPF_VALUE_TUPLE], Optional[int], Optional[int]], 'ComplexInfinity']` (commented at the following line) and some variants of it, but they couldn't reflect the fact that if the first argument of the tuple is `None` then the third must also be `None`, and sometimes a temp_result is impossible to be `ComplexInfinity` but mypy still requires to check this case. Dealing with these cases neither helps to fix the bug that this PR aims to fix, nor helps understand and refactor the code in evalf.py (at least now).
Moreover, `TEMP_RESULT` provides a way that unifys the return type of the `evalf`s. So imho it's better than nothing or tons of `Any`s here.

Reply:
If `TEMP_RESULT` is just an alias for `Any` then how is it better than `Any`?

Note that my main point in the comment above is that there should be comments in the code explaining what these types mean and why they are defined in this way.

Reply:
> If `TEMP_RESULT` is just an alias for `Any` then how is it better than `Any`?

Because you can not tell 2 `Any`s are actually the same type or not, but alias convey the information that they are the same type.

Reply:
> Note that my main point in the comment above is that there should be comments in the code explaining what these types mean and why they are defined in this way.

`An mpf value tuple is a tuple of integers (sign, man, exp, bc) ...` clearly explains why `MPF_VALUE_TUPLE = tTuple[int, int, int, int]` and what these 4 `int`s are. In fact I don't quite understand what scaled_zero tuple is, this name already appears in the comment, confusing me, so I simply add type hints for it.

Reply:
It's okay to write a comment in the code saying that you don't understand something. That kind of comment is actually very helpful to other readers,